fuel/datasets/hdf5.py

```diff
@@ -173,6 +174,7 @@ def __init__(self, file_or_path, which_sets, subset=None,
             self.external_file_handle = file_or_path
         else:
             self.path = file_or_path
+            self.path = dataset_cache.cache_file(self.path)
```

Comment:
I assume that we should also change the way the files are opened in all other basic dataset classes.